### PR TITLE
Err out better when failing to eval CEL

### PR DESCRIPTION
**Pull request by alexsnaps**

### CODE REVIEW

File: src/data/cel.rs [hunk 1]
```diff
@@ -55,7 +55,7 @@ impl Expression {
         Self::new_expression(expression, true)
     }
 
-    pub fn eval(&self) -> Value {
+    pub fn eval(&self) -> Result<Value, String> {
```
Comment (alexsnaps) [inline src/data/cel.rs:58]: This is the only real change really...

File: src/configuration/action.rs [hunk 2]
```diff
@@ -31,7 +31,16 @@ impl Action {
...
             self.conditions.is_empty() || self.conditions.iter().all(PatternExpression::applies)
         } else {
-            predicates.iter().all(Predicate::test)
+            predicates
+                .iter()
+                .enumerate()
+                .all(|(pos, predicate)| match predicate.test() {
+                    Ok(b) => b,
+                    Err(err) => {
+                        error!("Failed to evaluate {}: {}", self.predicates[pos], err);
```
Comment (alexsnaps) [inline src/configuration/action.rs:40]: ... and these are the real things that I wanted...

File: src/configuration/action.rs [hunk 3]
```diff
@@ -60,14 +69,20 @@ impl Action {
...
+                            Value::UInt(n) => format!("{n}"),
+                            Value::Float(n) => format!("{n}"),
+                            // todo this probably should be a proper string literal!
+                            Value::String(s) => (*s).clone(),
+                            Value::Bool(b) => format!("{b}"),
+                            Value::Null => "null".to_owned(),
+                            _ => panic!("Only scalar values can be sent as data"),
+                        },
+                        Err(err) => {
+                            error!("Failed to evaluate {}: {}", cel.value, err);
```
Comment (alexsnaps) [inline src/configuration/action.rs:83]: ... and this ...

File: src/configuration/action_set.rs [hunk 4]
```diff
@@ -51,7 +52,19 @@ impl ActionSet {
...
-            predicates.iter().all(Predicate::test)
+            predicates
+                .iter()
+                .enumerate()
+                .all(|(pos, predicate)| match predicate.test() {
+                    Ok(b) => b,
+                    Err(err) => {
+                        error!(
+                            "Failed to evaluate {}: {}",
+                            self.route_rule_conditions.predicates[pos], err
```
Comment (alexsnaps) [inline src/configuration/action_set.rs:63]: ... and this.

**Review by adam-cattermole — APPROVED** [review]

**Pull request merged**